PayloadPkg/OsLoader/LoadImage.c

```diff
@@ -152,12 +165,24 @@ GetBootImageFromRawPartition (
   //
   // Read the rest of the IAS image into the buffer
   //
-  Status = MediaReadBlocks (
-             BootOption->HwPart,
-             LogicBlkDev.StartBlock + LbaAddr + (AlignedHeaderSize / BlockSize),
-             (AlignedImageSize - AlignedHeaderSize),
-             (VOID *)((UINTN)Buffer + AlignedHeaderSize)
-             );
+
+  if ( BootOption->DevType == OsBootDeviceMemory )
+  {
+    Status = MediaReadBlocks (

```

Comment:
Simplify the code by using a new varaible for LbaAddr to call for MediaReadBlocks so that single function call can be used instead.